src/allmydata/test/web/test_status.py

```diff
@@ -0,0 +1,152 @@
+# Tests for code in allmydata.web.status
+
+from bs4 import BeautifulSoup
+from twisted.web.template import flattenString
+from zope.interface import implementer
+
+from allmydata.interfaces import (
+    IDownloadResults,
+    IDownloadStatus,
+)
+from allmydata.web.status import DownloadStatusElement
+
+from .common import (
+    assert_soup_has_favicon,
+    assert_soup_has_text,
+    assert_soup_has_tag_with_content,
+)
+from ..common import TrialTestCase
+
+
+@implementer(IDownloadResults)
+class FakeDownloadResults(object):
+
+    def __init__(self,
+                 file_size=0,
+                 servers_used=None,
+                 server_problems=None,
+                 servermap=None,
+                 timings=None):
+        """
+        See IDownloadResults for parameters.
+        """
+        self.file_size = file_size
+        self.servers_used = servers_used
+        self.server_problems = server_problems
+        self.servermap = servermap
+        self.timings = timings
+
+
+@implementer(IDownloadStatus)
+class FakeDownloadStatus(object):
+
+    def __init__(self,
+                 storage_index = None,
+                 file_size = 0,
+                 servers_used = None,
+                 server_problems = None,
+                 servermap = None,
+                 timings = None):
+        """
+        See IDownloadStatus and IDownloadResults for parameters.
+        """
+        self.storage_index = storage_index
+        self.file_size = file_size
+        self.dyhb_requests = []
+        self.read_events = []
+        self.segment_events = []
+        self.block_requests = []
+
+        self.servers_used = servers_used
+        self.server_problems = server_problems
+        self.servermap = servermap
+        self.timings = timings
+
+    def get_started(self):
+        return None
+
+    def get_storage_index(self):
+        return self.storage_index
+
+    def get_size(self):
+        return self.file_size
+
+    def using_helper(self):
+        return False
+
+    def get_status(self):
+        return "FakeDownloadStatus"
+
+    def get_progress(self):
+        return 0
+
+    def get_active():
+        return False
+
+    def get_counter():
+        return 0
+
+    def get_results(self):
+        return FakeDownloadResults(self.file_size,
+                                   self.servers_used,
+                                   self.server_problems,
+                                   self.servermap,
+                                   self.timings)
+
+# Tests for code in allmydata.web.status.DownloadStatusElement
```

Comment:
This comment should be the class docstring instead.

Reply:
Made this change in fe07ab3.